.gitignore

```diff
@@ -10,7 +10,7 @@ webapp/yarn-error.log
 mattermost.mattermost-license
 config/mattermost.mattermost-license
 config/config.json
-
+config/default.json
```

Comment:
Perhaps the default.json should still be included in the repository? The other generated code gets checked in, so it might make sense to keep this one in as well

Reply:
If the generated code is consumed by other code, I’d agree, but I note that this will now purely be a build byproduct. The difficulty if we keep it is that it then runs the risk of being out of sync. Though I guess we could a CI check for that?

Reply:
I'd be fine if all of the code is generated as long as it's done automatically. Without checking this in, we should make sure that the file is automatically generated both when running `make run-server` and `make package` (or whatever the command to build the bundle is). As far as I can tell, that doesn't happen right now though

Reply:
The CI steps might need a bit of tweaking: we don't want `default.json` generated, per se, but rather a default `config.json` as the CI server would normally effect by invoking `mv default.json config.json`. I think all we need to do is update the various `build/Jenkinsfile*` files to invoke `make config-reset`. @reflog, can you add that?

Reply:
@reflog, I guess we can remove this line now, since we don't actually generate `default.json` anywhere?

Reply:
I think it's better to leave it in case someone has default.json laying around and commits it by mistake...

Reply:
For purely generated files, I'd agree, but prior to this commit, this file was part of the repository. We don't add any `git rm`'ed file to `.gitignore` just in case someone still has it left around, so I think the same rules should apply to this file.

Anyone syncing to this commit will either have their file deleted, or be forced to resolve conflicts through the delete -- and then trying to submit it again would be a mistake like re-adding any other. deleted file.